openadapt/db.py

```diff
@@ -51,3 +52,101 @@ def get_base(engine):
 engine = get_engine()
 Base = get_base(engine)
 Session = sessionmaker(bind=engine)
+
+
+def copy_recording_data(
+    source_engine: sa.engine,
+    target_engine: sa.engine,
+    recording_id: int,
+    exclude_tables: tuple = (),
+) -> str:
+    """Copy a specific recording from the source database to the target database.
+
+    Args:
+        source_engine (create_engine): SQLAlchemy engine for the source database.
+        target_engine (create_engine): SQLAlchemy engine for the target database.
+        recording_id (int): The ID of the recording to copy.
+        exclude_tables (tuple, optional): Tables excluded from copying. Defaults to ().
+
+    Returns:
+        str: The URL or path of the target database.
+    """
+    try:
+        src_metadata = MetaData()
+        tgt_metadata = MetaData()
+
+        @event.listens_for(src_metadata, "column_reflect")
+        def genericize_datatypes(inspector, tablename, column_dict):
+            column_dict["type"] = column_dict["type"].as_generic(allow_nulltype=True)
+
+        src_conn = source_engine.connect()
+        tgt_conn = target_engine.connect()
+        tgt_metadata.reflect(bind=target_engine)
+        src_metadata.reflect(bind=source_engine)
+
+        # Drop all tables in target database
+        for table in reversed(tgt_metadata.sorted_tables):
+            if table.name not in exclude_tables:
+                print("Dropping table =", table.name)
```

Comment:
Please use logger.info

Reply:
Addressed in https://github.com/OpenAdaptAI/OpenAdapt/pull/223